Doc/bernede1/debug uberenv ci (#18)
LLNL/radiuss-ci: docs/sphinx/uberenv.rst

  definition, it has been vetted. It also ensures that this script is usable in
  interactive mode, making it easier to test.

Debugging
=========

In the workflow described above, there are 4 levels of scripts to control the
build of a package. From the lower to the higher level:

* The *build system* is controlled by the host-config file (generated by Spack
  or not).
* The *Spack package* is controlled by the spec provided and the configuration.
* *Uberenv* takes a spec and a json configuration file.

> * A *build_and_test script* also sometimes called test driver. Umpire and RAJA
>   ones require a spec and some other control variables.

Author: ```suggestion
* A *build_and_test script* also sometimes called test driver. The one in Umpire
  and RAJA requires a spec and some other control variables.
```

Now, when it comes to debugging, each level has some requirements to reproduce
an failing build:

* The *build_and_test script* typically runs in CI context. This means that it
  may not be designed to run outside CI. It is better if it does, and we try to
  do that in RADIUSS, but it is not guaranteed.
* *Uberenv* is meant to provide a turnkey way to install the project and its
  dependencies. It is usually a good way to reproduce a build on the same

>   machine. The CI works with working directories where the uberenv install

Author: ```suggestion
  machine. The CI creates working directories in which the uberenv install
```
